data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/threshold/ThresholdCheck.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.threshold;
+/**
+ * ThresholdCheck receives paramaters for which to reference the
+ * limits of a buffer and CloudWatchLogsClient before making a
+ * PutLogEvent request to AWS.
+ */
+public class ThresholdCheck {
```

Comment:
This class appears to be unused (aside from unit tests). Why does it have so many public methods? Will the calling code need this level of granularity?

For example, why do we need to know equal versus greater than? Why not combine them?

Also, why does the caller need to know which threshold is reached (e.g. bytes vs. interval vs. event size)?

Reply:
The main reason for the granularity is for the limits for sending data through the CloudWatchLogs SDK, as the API has byte size limitations and to be as strict with the calls to it, we need to be able to define a hard threshold (with byte size and batch size). This way the customer has control over how large the request to the API can be.

Reply:
I have made most of the internal methods private as they are not actually necessary to the caller, thank you. The only ones that are public are checkMaxEventSize(), as this is more of an error checking method that is used prior to thresholding.